javascripts/dark-mode.js

```diff
@@ -57,55 +57,66 @@ define([], () => {
-    lightModeEnabled = value === 'light';
-  }
-
-  function setupDarkModeListener() {
-    var r = document.querySelector(':root');
-    if (!r) {
-      return;
-    }
+  /* eslint block-scoped-var: "off" */
 
+// Wrap your code in a function to handle possible async operations
+async function setupDarkModeListener() {
+  // Use a try-catch block for error handling
+  try {
+    const root = document.documentElement;
     const viewModeElement = document.getElementById('view-mode');
-    if (!viewModeElement) {
-      return;
+    const viewModeAnchor = document.getElementById('view-mode-a');
+    const goBackHomeElement = document.getElementById('go-back-home');
+
+    if (!root || !viewModeElement || !viewModeAnchor || !goBackHomeElement) {
+      throw new Error('One or more required elements not found.');
     }
 
-    const viewModeAnchor = document.getElementById('view-mode-a');
-    if (!viewModeAnchor) {
-      return;
+    const storedValue = window.sessionStorage.getItem('mode');
+    let lightModeEnabled = !(storedValue && storedValue === 'dark');
+
+    function setDarkMode() {
+      root.style.setProperty('--body-back', '#1a2025');
+      // Add more style property assignments here for dark mode
+
+      // Update the image sources and title attributes
+      goBackHomeElement.setAttribute('src', '/images/logo_dark_1.png');
+      viewModeElement.setAttribute('src', '/images/sun-light.png');
+      viewModeAnchor.title = 'light-mode';
     }
 
-    const goBackHomeAnchor = document.getElementById('go-back-home-a');
-    if (!goBackHomeAnchor) {
-      return;
+    function setLightMode() {
+      root.style.setProperty('--body-back', '#f9f9f9');
+      // Add more style property assignments here for light mode
+
+      // Update the image sources and title attributes
+      goBackHomeElement.setAttribute('src', '/images/logo.png');
+      viewModeElement.setAttribute('src', '/images/Dim-Night.png');
+      viewModeAnchor.title = 'dark-mode';
     }
 
-    const goBackHomeElement = document.getElementById('go-back-home');
-    if (!goBackHomeElement) {
-      return;
+    function updateValue(value) {
+      window.sessionStorage.setItem('mode', value);
+      lightModeEnabled = value === 'light';
     }
 
     if (!lightModeEnabled) {
-      setDarkMode(viewModeElement, viewModeAnchor, goBackHomeElement);
+      setDarkMode();
     }
 
     viewModeAnchor.addEventListener('click', () => {
       if (lightModeEnabled) {
-        setDarkMode(viewModeElement, viewModeAnchor, goBackHomeElement);
+        setDarkMode();
         updateValue('dark');
       } else {
-        setLightMode(viewModeElement, viewModeAnchor, goBackHomeElement);
+        setLightMode();
         updateValue('light');
       }
     });
+  } catch (error) {
+    console.error('Error in setupDarkModeListener:', error);
   }
+}
 
-  return setupDarkModeListener;
-});
+// Export the setupDarkModeListener function
+export default setupDarkModeListener;
```

Comment:
I'm not a fan of exporting arbitrarily, and when I test this out on our staging site I see an error here: https://deploy-preview-4248--up-for-grabs-test-bench.netlify.app/

![](https://github.com/up-for-grabs/up-for-grabs.net/assets/359239/997cb65a-ed4b-4665-87ec-2a74b20ed714)

Reply:
There's a reason why these scripts are authored in a specific way, and rewriting these to ES Modules is not something I have explored: https://github.com/up-for-grabs/up-for-grabs.net/blob/gh-pages/docs/client-side-scripting.md